lib/rmt.toit

```diff
@@ -356,59 +412,117 @@ class Channel:
 
   is_configured -> bool:
-    return configured_ != 0
+    return configured_ != NOT_CONFIGURED_
 
   is_input -> bool:
-    return configured_ == 1
+    return configured_ == CONFIGURED_AS_INPUT_
 
   is_output -> bool:
-    return configured_ == 2
+    return configured_ == CONFIGURED_AS_OUTPUT_
 
   idle_threshold -> int?:
     return rmt_get_idle_threshold_ resource_
 
   idle_threshold= threshold/int -> none:
+    if not 1 <= threshold <=0x7FFF: throw "INVALID_ARGUMENT"
     rmt_set_idle_threshold_ resource_ threshold
 
+  is_reading -> bool:
+    return is_reading_
+
   /**
-  Returns the buffer size in bytes.
+  Starts receiving signals for this channel.
+
+  This channel must be configured for receiving (see $(configure --input)).
 
-  This value is null if the channel is not configured for input.
+  If $flush is set (the default) flushes all buffered signals.
+  */
+  start_reading --flush/bool=true -> none:
+    if not is_input: throw "INVALID_STATE"
+    is_reading_ = true
+    rmt_start_receive_ resource_ flush
+
+  /**
+  Stops receiving.
   */
-  buffer_size -> int?:
-    return rx_buffer_size_
+  stop_reading -> none:
+    is_reading_ = false
+    rmt_stop_receive_ resource_
 
   /**
   Receives signals.
 
   This channel must be configured for receiving (see $(configure --input)).
 
-  If $max_bytes is not sufficient to store all received signals, then the
-    result is truncated. If it is important to detect this condition, then the user should
-    set $max_bytes to a value greater than the maximum expected signal byte size.
+  The result may contain trailing 0-period signals. Those should be ignored.
 
-  Warning: currently the receiver is blocking which may cause multiple issues:
-  - the watchdog timer might be triggered
-  - no global garbage collection can run while the receiver is waiting.
+  If the channel hasn't yet started to read, starts reading ($start_reading).
+    However, does not flush
 
-  The $timeout_ms must be big enough for the RMT peripheral to copy the data into its
-    internal buffer. At the very least the timeout thus must be bigger than the idle threshold.
-    However, because of context switches etc, several milliseconds are usually required
-    before the data is available in the internal buffers.
+  If $stop_reading is true, stops reading after the next signal is received.
+  If $stop_reading is false, always keeps the channel reading.
+  If $stop_reading is null, stops reading if the channel wasn't reading yet.
   */
-  read max_bytes/int --timeout_ms/int=DEFAULT_READ_TIMEOUT_MS -> Signals:
+  read --stop_reading/bool?=null -> Signals:
     if not is_input: throw "INVALID_STATE"
-    if max_bytes > buffer_size: throw "maximum returned buffer size greater than allocated buffer size"
-    bytes := rmt_receive_ resource_ max_bytes timeout_ms
-    return Signals.from_bytes bytes
+
+    was_reading := is_reading_
+    if stop_reading == null: stop_reading = not was_reading
+
```

Comment:
I'd move the newline onto the other side of the 'if not was_reading' line.

Reply:
done.